spec/unit/plugins/windows/filesystem_spec.rb

```diff
@@ -194,33 +194,33 @@
   end
 
   describe "#merge_info" do
-    let(:info1) do
-      { "drive1" => { "x" => 10, "y" => "test1" },
-        "drive2" => { "x" => 20, "z" => "test2" } }
+    let(:logical_info) do
+      { "dev1,drive1" => { "mount" => "drive1", "x" => 10, "y" => "test1" },
+        "dev2,drive2" => { "mount" => "drive2", "x" => 20, "z" => "test2" },
+        "dev2,drive3" => { "mount" => "drive3", "x" => 20, "z" => "test3" } }
     end
-    let(:info2) do
+    let(:encryption_info) do
       { "drive1" => { "k" => 10, "l" => "test1" },
         "drive2" => { "l" => 20, "m" => "test2" } }
     end
-    let(:info3) { { "drive1" => { "o" => 10, "p" => "test1" } } }
-    let(:info4) { { "drive2" => { "q" => 10, "r" => "test1" } } }
-
-    it "Returns an empty mash when no info is passed" do
-      expect(plugin.merge_info([])).to be_a(Mash)
-      expect(plugin.merge_info([])).to be_empty
-    end
+    let(:logical_info2) { { ",drive1" => { "mount" => "drive1", "o" => 10, "p" => "test1" } } }
+    let(:encryption_info2) { { "drive2" => { "q" => 10, "r" => "test1" } } }
```

Comment:
I don't think so... that doesn't work anymore, which is why I broke that apart. AIUI, `encryption_info` **only** returns information on the mountpoint (i.e. drive letter), not volume. So imagine that you had:

`MyCoolVolume` mounted ad `E:`
EncryptionInfo doesn't know what MyCoolVolume is because the encryption commands don't return it, they only return the drive letter (E:)

So in order to safely build data we build by pair, so we build a key like `MyCoolVolume,E:`. However, if we were to try to pretend the `encryption_info` could be the same, then we'd build `,E:`... and that doesn't match. Pretending it does is a lie. To "merge" these, we'd always ignore the first, part, look at the second part and then do exactly what I do above, look at anything that has a `mount` of `E:`.

Meanwhile if we don't have a VolumeName then `logical_info` can of course return `,E:`, which has a specific meaning. It means a null-named volume is mounted at `E:`. But thats **not** what `encryption_info` has ever been able to tell you - all it can tell you is some encryption info about **whatever** happens to be mounted at `E:`.

Therefore trying to make the keys the same is impossible, and making them *look* the same is misleading and confusing, so I very specifically made it clear they were saying different things.